src/legacy/core_plugins/data/public/search/search_service.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Plugin, CoreSetup, CoreStart } from '../../../../../core/public';
+import {
+  aggTypes,
+  AggType,
+  AggConfig,
+  AggConfigs,
+  FieldParamType,
+  MetricAggType,
+  aggTypeFieldFilters,
+  setBounds,
+  parentPipelineAggHelper,
+  siblingPipelineAggHelper,
+} from './aggs';
+
+interface AggsSetup {
+  types: typeof aggTypes;
+}
+
+interface AggsStart {
+  types: typeof aggTypes;
+  AggConfig: typeof AggConfig;
+  AggConfigs: typeof AggConfigs;
+  AggType: typeof AggType;
+  aggTypeFieldFilters: typeof aggTypeFieldFilters;
+  FieldParamType: typeof FieldParamType;
+  MetricAggType: typeof MetricAggType;
+  parentPipelineAggHelper: typeof parentPipelineAggHelper;
+  siblingPipelineAggHelper: typeof siblingPipelineAggHelper;
+  setBounds: typeof setBounds;
+}
+
+export interface SearchSetup {
+  aggs: AggsSetup;
+}
+
+export interface SearchStart {
+  aggs: AggsStart;
+}
+
+/**
+ * The contract provided here is a new platform shim for ui/agg_types.
+ *
+ * Once it has been refactored to work with new platform services,
+ * it will move into the existing search service in src/plugins/data/public/search
+ */
+export class SearchService implements Plugin<SearchSetup, SearchStart> {
```

Comment:
just a question, should SearchService implement a Plugin interface? If I understand properly it will be a part of Data plugin, not a separate plugin.

Reply:
ah, yes, it will live in the data plugin. but technically the way we author many of our service definition files is compatible with the `Plugin` generic, so in this case TS is still happy. 

there has been discussion in the past about implementing a `Service` interface for these types of things. for now i can remove to avoid confusion.